XUtils.h

```diff
@@ -56,6 +56,17 @@ void String_freeArray(char** s);
 
 char* String_readLine(FILE* fd) ATTR_MALLOC;
 
+static inline char *String_charNull(const char* s, int c) {
+#ifdef HAVE_STRCHRNUL
+   return strchrnul(s, c);
+#else
+   char *result;
+   if ((result = strchr(s, c)))
+      return result;
+   return strchr(s, '\0');
```

Comment:
I have a naïve implementation here, because I don't think calling `strchr` twice is a good idea for performance:
```c
   char* p = (char*)s;
   while (*p && *p != c)
      p++;
   return p;
```

Reply:
OTOH, strchr is highly optimised so chances are fair-to-middling that two calls are still more efficient than a naive re-implementation.  Think I'll stick with the strchr version unless measurements show compelling evidence to the contrary.